.github/workflows/containers.yml

```diff
@@ -983,3 +983,53 @@ jobs:
                         integration-tests/build/reports/**
 
             -   uses: ./.github/actions/teardown
+
+    E2EUITests:
+        needs: PublishJibContainers
+        runs-on: ubuntu-latest
+        container: cypress/browsers:node14.17.0-chrome91-ff89
+        timeout-minutes: 10
+
+        services:
+            api-catalog-services:
+                image: ghcr.io/balhar-jakub/api-catalog-services:${{ github.event.pull_request.number || 'latest' }}
+                volumes:
+                    - /api-defs:/api-defs
+            discoverable-client:
+                image: ghcr.io/balhar-jakub/discoverable-client:${{ github.event.pull_request.number || 'latest' }}
+            discovery-service:
+                image: ghcr.io/balhar-jakub/discovery-service:${{ github.event.pull_request.number || 'latest' }}
+                volumes:
+                    - /api-defs:/api-defs
+            gateway-service:
+                image: ghcr.io/balhar-jakub/gateway-service:${{ github.event.pull_request.number || 'latest' }}
+            mock-services:
+                image: ghcr.io/balhar-jakub/mock-services:${{ github.event.pull_request.number || 'latest' }}
+
+        steps:
+            -   uses: actions/checkout@v2
+                with:
+                    ref: ${{ github.head_ref }}
+            -   uses: ./.github/actions/setup
+            -   name: Install npm dependencies
+                uses: bahmutov/npm-install@v1
+                with:
+                    working-directory: api-catalog-ui/frontend
+            -   name: Cache NPM and Cypress 📦
+                uses: actions/cache@v2
+                with:
+                    path: |
+                        ~/.cache/Cypress
+                        node_modules
+                    key: my-cache-${{ runner.os }}-${{ hashFiles('package-lock.json') }}
+            -   name: Cypress run
+                run: |
+                    cd api-catalog-ui/frontend
+                    npm run cy:e2e:ci
+            -   name: Upload screenshots
+                uses: actions/upload-artifact@v2
+                if: failure()
+                with:
+                    name: cypress-snapshots
+                    path: api-catalog-ui/frontend/cypress/screenshots
+
```

Comment:
Should call `-   uses: ./.github/actions/teardown` at end of action.

Reply:
Good catch, fixed